apps/ca.c

```diff
@@ -940,8 +940,10 @@ int ca_main(int argc, char **argv)
             if (j > 0) {
                 total_done++;
                 BIO_printf(bio_err, "\n");
-                if (!BN_add_word(serial, 1))
+                if (!BN_add_word(serial, 1)) {
+                    X509_free(x);
                     goto end;
+                }
                 if (!sk_X509_push(cert_sk, x)) {
                     BIO_printf(bio_err, "Memory allocation failure\n");
```

Comment:
Shouldn't we also free `x` here too?

Reply:
Yes, you're right.

Reply:
Some changes are required to apply it to 1.1.0 and 1.0.2. I'll make appropriate pull requests.